app/models/release.rb

```diff
@@ -92,9 +92,12 @@ def next_number(current_version)
     current_version.to_s.dup.sub!(/\d+$/) { |d| d.to_i + 1 }
   end
 
-  def covert_ref_to_sha
+  def convert_ref_to_sha
     return if commit.blank? || commit =~ Build::SHA1_REGEX
-    self.commit = project.repository.commit_from_ref(commit)
+    tag = GITHUB.compare(project.repository_path, commit, commit)
```

Comment:
please use `Project#repo_commit_from_ref`

Reply:
The comment from that function
>   # quick commit from reference without pulling the repository ... unless we have no remote

The issue we're trying to resolve is, when there's no cached repository then the webhook request process will take a long time (and timeout).
AFAICT this is still part of that request, as we create the Release as part of that.

So I don't think using that function will solve this.